src/openrct2-ui/windows/LoadSave.cpp

```diff
@@ -962,10 +962,9 @@ static void WindowLoadsaveInvokeCallback(int32_t result, const utf8* path)
     }
 }
 
-static void SavePath(utf8** config_str, const char* path)
+static void SavePath(u8string* config_str, u8string_view path)
```

Comment:
Wouldn't it better to just return the result?

Reply:
It would be, but this function also triggers a config save so assignment needs to be done **before** that.

Reply:
Is that reference pointing to a global? It could simply pass the argument to config_save_default if so.

Reply:
I'm not sure what do you mean - it points to different config entries from the global scope, e.g. `gConfigGeneral.last_save_game_directory` or `gConfigGeneral.last_save_landscape_directory`

Reply:
What I mean is
```cpp
static u8string SavePath(u8string_view path)
{
    config_save_default();
	return Path::GetDirectory(path);
}
```
Or what I'm not seeing?

Reply:
I'm pretty sure this will modify the global variable after saving, whilst it has to be modified before saving.

Reply:
Which variable?

Reply:
I just checked the actual code, the config_save_default call in SavePath seems like a horrible idea to me, the function name "SavePath" is also not optimal.

Reply:
I changed the argument to a reference and renamed the function to `SetAndSaveConfigPath` - I think it is still clear what it does but it's also shorter.

Reply:
Well one could argue that its not clear if you set the config path or one of the paths in the config 🙃 But I think its fine for now, avoiding the crashes is more important.

Reply:
I don't disagree, but the way the function is used (accepting 2 parameters) makes it clearer IMO.